tests/singlenode/singlenode_test.go

```diff
@@ -0,0 +1,67 @@
+package test
+
+import (
+	"github.com/ashraffouda/grid-provider/tests"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"os/exec"
+	"testing"
+)
+
+func TestSingleNodeDeployment(t *testing.T) {
+	/* Test case for deployeng a singlenode.
+
+	   **Test Scenario**
+
+	   - Deploy a singlenode.
+	   - Check that the outputs not empty.
+	   - Up wireguard.
+	   - Check that containers is reachable
+	   - Destroy the deployment
+
+	*/
+
+	// retryable errors in terraform testing.
+	publicKey := os.Getenv("PUBLICKEY")
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "./",
+		Vars: map[string]interface{}{
+			"public_key": publicKey,
+		},
+		Parallelism: 1,
+	})
+
+	terraform.InitAndApply(t, terraformOptions)
+	defer terraform.Destroy(t, terraformOptions)
+
+	// Check that the outputs not empty
+	publicIP := terraform.Output(t, terraformOptions, "public_ip")
+	assert.NotEmpty(t, publicIP)
+
+	node1Container1IP := terraform.Output(t, terraformOptions, "node1_container1_ip")
+	assert.NotEmpty(t, node1Container1IP)
+
+	node1Container2IP := terraform.Output(t, terraformOptions, "node1_container2_ip")
+	assert.NotEmpty(t, node1Container2IP)
+
+	// Up wireguard
+	wgConfig := terraform.Output(t, terraformOptions, "wg_config")
+	assert.NotEmpty(t, wgConfig)
+	tests.UpWg(wgConfig)
+	defer tests.DownWG()
+
+	// Check that containers is reachable
+	out, _ := exec.Command("ping", publicIP, "-c 5", "-i 3", "-w 10").Output()
+	assert.NotContains(t, string(out), "Destination Host Unreachable")
+
+	out1, _ := exec.Command("ping", node1Container2IP, "-c 5", "-i 3", "-w 10").Output()
+	assert.NotContains(t, string(out1), "Destination Host Unreachable")
+
+	out2, _ := exec.Command("ping", node1Container1IP, "-c 5", "-i 3", "-w 10").Output()
+	assert.NotContains(t, string(out2), "Destination Host Unreachable")
+
+	// ssh to container
+	_, errors := tests.RemoteRun("root", node1Container1IP, "ls")
```

Comment:
and the other node? you deployed 2 VMs

Reply:
make sure to verify the VMs that they match the specs and the flist
![image](https://user-images.githubusercontent.com/64129/132703115-de9a3617-a3fd-47e2-870e-c257007f3e8a.png)

verify on the env variables passed as well
verify on the ips to match within the vms

Apply that comment to all tests